components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/util/RegistrationFlowEngineUtils.java

```diff
@@ -103,12 +113,16 @@ public static void removeRegContextFromCache(String contextId) {
     /**
      * Initiate the registration context.
      *
-     * @param tenantDomain Tenant domain.
+     * @param tenantDomain  Tenant domain.
+     * @param callbackUrl   Callback URL.
+     * @param applicationId Application ID.
      * @return Registration context.
+     * @throws RegistrationEngineException Registration framework exception.
      */
-    public static RegistrationContext initiateContext(String tenantDomain)
+    public static RegistrationContext initiateContext(String tenantDomain, String callbackUrl, String applicationId)
             throws RegistrationEngineException {
 
+        RegistrationContext context = new RegistrationContext();
```

Comment:
shall we move context object initialization and return statement both into the try block for better readability